src/compile/data/geojson.ts

```diff
@@ -0,0 +1,48 @@
+import {SHAPE, X, X2, Y, Y2} from '../../channel';
+import {GEOJSON, LATITUDE, LONGITUDE} from '../../type';
+import {contains, duplicate, keys} from '../../util';
+import {VgGeoJSONTransform} from '../../vega.schema';
+import {ModelWithField} from '../model';
+import {DataFlowNode} from './dataflow';
+
+export class GeoJSONNode extends DataFlowNode {
+  public clone() {
+    return new GeoJSONNode(duplicate(this.fields), this.geojson, this.signal);
+  }
+
+  public static make(model: ModelWithField): GeoJSONNode {
+    const geo = model.reduceFieldDef((rel, fieldDef, channel) => {
+      if ((contains([X, Y, X2, Y2], channel) && contains([LATITUDE, LONGITUDE], fieldDef.type))
+        || (channel === SHAPE && fieldDef.type === GEOJSON)) {
+        rel[fieldDef.type] = fieldDef.field;
```

Comment:
I wonder if using dictionary here would make it work when we want to use both x/y and y2/x2 (e.g., for line segment rules)?   

It seems like this logic will make fieldDef for x2 overrides x's

Reply:
you've discovered a bug, but unfortunately it's not so easy to fix. The bug isn't about whether or not we can project `rule`, but instead how `fit` is computed (it appears to work since `fit` on X2 and Y2 is similar to X and Y). You're correct in your assessment that X2 would override X (my bad).

Reply:
this is non trivial. I will have to think about it more.

Reply:
Resolved.